ruby/lib/gilded_rose.rb

```diff
@@ -1,68 +1,52 @@
+require "items"
+require "item_type"
+require "helper_functions"
+
 class GildedRose
+  attr_reader :items
+
+  SULFURAS = "Sulfuras, Hand of Ragnaros"
+  BRIE = "Aged Brie" 
+  BACKSTAGE = "Backstage passes to a TAFKAL80ETC concert" 
 
   def initialize(items)
     @items = items
   end
 
-  def update_quality()
-    @items.each do |item|
-      if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert"
-        if item.quality > 0
-          if item.name != "Sulfuras, Hand of Ragnaros"
-            item.quality = item.quality - 1
-          end
-        end
-      else
-        if item.quality < 50
-          item.quality = item.quality + 1
-          if item.name == "Backstage passes to a TAFKAL80ETC concert"
-            if item.sell_in < 11
-              if item.quality < 50
-                item.quality = item.quality + 1
-              end
-            end
-            if item.sell_in < 6
-              if item.quality < 50
-                item.quality = item.quality + 1
-              end
-            end
-          end
-        end
-      end
-      if item.name != "Sulfuras, Hand of Ragnaros"
-        item.sell_in = item.sell_in - 1
-      end
-      if item.sell_in < 0
-        if item.name != "Aged Brie"
-          if item.name != "Backstage passes to a TAFKAL80ETC concert"
-            if item.quality > 0
-              if item.name != "Sulfuras, Hand of Ragnaros"
-                item.quality = item.quality - 1
-              end
-            end
-          else
-            item.quality = item.quality - item.quality
-          end
-        else
-          if item.quality < 50
-            item.quality = item.quality + 1
-          end
-        end
-      end
+  def update_quality
+    items.each do |item|
+      update_item(item)
+    end
+  end
+
+  def update_item(item)
+    if item.name == SULFURAS
+    elsif item.name == BRIE
+      aged_brie(item)
+    elsif item.name == BACKSTAGE
+      backstage_passes(item)
+    else
+      normal_item(item)
     end
   end
 end
 
-class Item
-  attr_accessor :name, :sell_in, :quality
+private
 
-  def initialize(name, sell_in, quality)
-    @name = name
-    @sell_in = sell_in
-    @quality = quality
-  end
+def increase_quality(item)
+  item.quality = item.quality + 1
+end
 
-  def to_s()
-    "#{@name}, #{@sell_in}, #{@quality}"
-  end
+def decrease_quality(item)
+  item.quality = item.quality - 1
+end
+
+def decrease_to_zero(item)
```

Comment:
Just watch name and indentation -- this is decrease quality, right?  

I love passing the item as an argument instead of relying, say, on some instance variable.

Reply:
`decrease_quality`...item.quality = 0